### PR TITLE
Add support for NTP options in Kubevirt.

### PR DESCRIPTION
**Pull request by justinbarrick**
Signed-off-by: Justin Barrick <justin.m.barrick@gmail.com>

**What this PR does / why we need it**:

This pull request adds support for configuring NTP servers for Kubevirt.

**Release note**:

```release-note
Add support for configuring NTP servers for VMs via NTP.
```

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically.
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by justinbarrick** [comment]
The errors in CI seem unrelated. Are they flakey or did I break something?

File: tests/vmi_networking_test.go [hunk 1]
```diff
@@ -568,6 +568,7 @@ var _ = Describe("Networking", func() {
 			dhcpVMI.Spec.Domain.Devices.Interfaces[0].DHCPOptions = &v1.DHCPOptions{
 				BootFileName:   "config",
 				TFTPServerName: "tftp.kubevirt.io",
+				NTPServers:     []string{"127.0.0.1"},
```
Comment (SchSeba) [inline tests/vmi_networking_test.go]: Can you please add multiple ntp servers here for the test?

Comment (justinbarrick) [reply tests/vmi_networking_test.go]: done

**Comment by SchSeba** [comment]
ci test please

**Comment by justinbarrick** [comment]
Is CI flakey?

**Comment by stu-gott** [comment]
@justinbarrick unfortunately it is a bit unstable currently. The failures do not appear related to your changes.

**Comment by justinbarrick** [comment]
It happens! Thanks for letting me know. Should we re-run?

File: pkg/virt-api/webhooks/validating-webhook/validating-webhook.go [hunk 2]
```diff
@@ -1076,6 +1076,17 @@ func ValidateVirtualMachineInstanceSpec(field *k8sfield.Path, spec *v1.VirtualMa
 				isVirtioNicRequested = true
 			}
 
+			if iface.DHCPOptions != nil {
+				for index, ip := range iface.DHCPOptions.NTPServers {
+					if net.ParseIP(ip).To4() == nil {
```
Comment (stu-gott) [inline pkg/virt-api/webhooks/validating-webhook/validating-webhook.go]: The wire format specified in the DHCP protocol mandates exactly 4 octets per server address for option 42 (RFC 1533, section 8.3), so I hope this code will never be exercised outside of unit tests. :)

**Comment by stu-gott** [comment]
ci test please

File: pkg/virt-api/webhooks/validating-webhook/validating-webhook.go (hunk 2 above)
Comment (justinbarrick) [reply pkg/virt-api/webhooks/validating-webhook/validating-webhook.go]: Yeah, this check enforces that it is a valid IPv4 address.

**Comment by justinbarrick** [comment]
Today is my last day at Cloudflare (the company that needs this), so I've handed this over to @gaahrdner to see through.

**Comment by gaahrdner** [comment]
Thanks @justinbarrick 

If you guys need any more information to get this merged please feel free to @me

**Pull request merged**